### ISSUE TITLE
Problem with form submit simulation

### ISSUE THREAD

**Issue by nithinpeter** [airbnb/enzyme#1858]
**Bug**
We have a `form` and a `submit button` outside the form. The button has `form` which is same as the form `id`. So whenever form gets submitted it will implicitly trigger the onClick handler on the button. 

I am trying to test this behaviour. I want to test if the onClick handler on the button gets called when I simulate `submit` on the form. But it is not working.

**To Reproduce**
Issue has been reproduced here:  https://codesandbox.io/s/jz91881vm5

** JS DOM? **
May it is a bug in JS-DOM?

**Comment by nithinpeter** [comment]
I agree. But for our specific use case we have to place the submit button outside the form. 
Browser triggers the click event on the submit button, when the form is submitted(by  hitting <enter> key). I am just wondering is there a way to write a test for this.

**Comment by ljharb** [comment]
https://jsfiddle.net/69z2wepo/311935/ shows that the browser may try to send a click event to the closest submit button, which won't necessarily be the outer one.

At any rate, this isn't something you should be testing with enzyme - because that's testing the DOM itself.

What you should be testing is that there's a submit button with the proper onClick, but you don't need to be testing the DOM, react itself, or the browser.
